src/components/settings-modal/settings-modal.js

```diff
@@ -40,12 +46,32 @@ const SettingsModal = ({
   const [isPrettyNameValue, setIsPrettyName] = useState(isPrettyName);
   const [showFeatureHintsValue, setShowFeatureHintsValue] =
     useState(showFeatureHints);
+  const [showDatasetPreviewsValue, setShowDatasetPreviewsValue] =
+    useState(showDatasetPreviews);
   const [toggleFlags, setToggleFlags] = useState(flags);
 
   useEffect(() => {
     setShowFeatureHintsValue(showFeatureHints);
   }, [showFeatureHints]);
 
+  useEffect(() => {
+    setShowDatasetPreviewsValue(showDatasetPreviews);
+  }, [showDatasetPreviews]);
+
+  useEffect(() => {
+    if (visible.settingsModal) {
```

Comment:
is this check required ? Also, I wonder how the toggle changes if the api takes time to return. I felt we can get the state before settingsModal is open. Based on the code, it seems the state is fetched after the modal is visible

Reply:
@ravi-kumar-pilla Yes `if (visible.settingsModal)` this check will make sure that we get preferences when we open it and not when we close. Current setup is fine as `visible.settingsModal` get true it make api call, and therefore `showDatasetPreviews` changes and this triggers `setShowDatasetPreviewsValue` to update. 

```
  useEffect(() => {
    setShowDatasetPreviewsValue(showDatasetPreviews);
  }, [showDatasetPreviews]);
```

```
  useEffect(() => {
    if (visible.settingsModal) {
      getPreferences();
    }
  }, [visible.settingsModal, getPreferences]);
```